pkg/kubectl/cmd/logs/logs.go

```diff
@@ -192,6 +192,7 @@ func (o *LogsOptions) ToLogOptions() (*corev1.PodLogOptions, error) {
 	}
 
 	if len(o.Selector) > 0 && o.Tail == -1 {
+	} else if len(o.Selector) > 0 && o.Tail < -1 {
```

Comment:
This is not a good solution, you should be checking whether a flag was changed, through its `.Changed` attribute in `Complete` method and set a bool flag to tell the difference.

Reply:
Hi soltysh,

Thanks for your reviewing. Please help review again. Thanks.

if len(o.Selector) > 0 && o.Tail == -1 && o.TailSpecified {
		logOptions.TailLines = nil  <====if user specify -1, it means show all lines
	} else if len(o.Selector) > 0 && o.Tail < -1 {
		logOptions.TailLines = &o.Tail  <===== tail must greater than -1
	} else if o.Tail != -1 {
		logOptions.TailLines = &o.Tail
	}

Regards,
Jishan

Reply:
I think you have the condition backwards.

Reply:
Hi soltysh,

Thanks for your review. 
I changed the code according to your suggestion. It will pass the -1 to the server side, so we need change the API's validation too.
   if o.TailSpecified {
        logOptions.TailLines = &o.Tail
    }
Please help review again. Thanks.